erpnext_shopify/sync_products.py

```diff
@@ -0,0 +1,482 @@
+		sync_item_with_shopify(item, price_list, warehouse)
+
+def sync_item_with_shopify(item, price_list, warehouse):
+	variant_item_code_list = []
+
+	item_data = { "product":
+		{
+			"title": item.get("item_name"),
+			"body_html": item.get("description"),
+			"product_type": item.get("item_group"),
+			"published_scope": "global",
+			"published_status": "published",
+			"published_at": datetime.datetime.now().isoformat()
+		}
+	}
+
+	if item.get("has_variants"):
+		variant_list, options, variant_item_code = get_variant_attributes(item, price_list, warehouse)
+
+		item_data["product"]["variants"] = variant_list
+		item_data["product"]["options"] = options
+
+		variant_item_code_list.extend(variant_item_code)
+
+	else:
+		item_data["product"]["variants"] = [get_price_and_stock_details(item, warehouse, price_list)]
+
+	erp_item = frappe.get_doc("Item", item.get("item_code"))
+
+	# check if the item really exists on shopify
+	if item.get("shopify_id"):
+		try:
+			get_request("/admin/products/{}.json".format(item.get("shopify_id")))
+		except requests.exceptions.HTTPError, e:
+			if e.args[0] and e.args[0].startswith("404"):
+				disable_shopify_sync_for_item(erp_item)
+				return
+			else:
+				disable_shopify_sync_for_item(erp_item)
+				raise
+			
+	if not item.get("shopify_id"):
+		new_item = post_request("/admin/products.json", item_data)
+		erp_item.shopify_id = new_item['product'].get("id")
+
+		if not item.get("has_variants"):
+			erp_item.shopify_variant_id = new_item['product']["variants"][0].get("id")
+
+		erp_item.save()
+
+		update_variant_item(new_item, variant_item_code_list)
+
+	else:
+		item_data["product"]["id"] = item.get("shopify_id")
+		put_request("/admin/products/{}.json".format(item.get("shopify_id")), item_data)
+				
+	sync_item_image(erp_item)
+
+def sync_item_image(item):
+	image_info = {
+        "image": {}
+	}
+
+	if item.image:
+		img_details = frappe.db.get_value("File", {"file_url": item.image}, ["file_name", "content_hash"])
+		if img_details and img_details[0] and img_details[1]:
+			is_private = item.image.startswith("/private/files/")
+			with open(get_files_path(img_details[0].strip("/"), is_private=is_private), "rb") as image_file:
+			    image_info["image"]["attachment"] = base64.b64encode(image_file.read())
+			image_info["image"]["filename"] = img_details[0]
+
+		elif item.image.startswith("http") or item.image.startswith("ftp"):
+			image_info["image"]["src"] = item.image
+
+		if image_info["image"]:
+			try:
+				if not exist_item_image(item.shopify_id, image_info):
+					try:
+						post_request("/admin/products/{0}/images.json".format(item.shopify_id), image_info)
+					except requests.exceptions.HTTPError, e:
+						if e.args[0] and e.args[0].startswith("422"):
+							disable_shopify_sync_for_item(item)
+						else:
+							disable_shopify_sync_for_item(erp_item)
+							raise
+						
+			except ShopifyError:
+				raise ShopifyError
+				
+def exist_item_image(shopify_id, image_info):
```

Comment:
This function is poorly named and does not clearly state what it does. Also you should add comments inside the function to explain what these if conditions mean.